### PR TITLE
Unable to retrieve media settings with just Vendor name

### PR DESCRIPTION
**Pull request by mihirpat1**
#### Description

If the media_settings.json file has media settings specified with just the vendor name and it doesn't include the vendor part number, the function get_media_settings_value fails to retrieve media settings.

#### Motivation and Context

It seems that after the PR [Add Port SI Configuration Per Speed](https://github.com/sonic-net/sonic-platform-daemons/pull/400) was merged, I am seeing a difference in behavior related to get_media_settings_value function.

Before the PR was merged, the function was able to populate media settings even if the vendor_name specified in the global profile was partially matching with the vendor_key (combination of vendor_name + vendor_pn). However, with the above PR merged, it is now mandatory to have the global profile with the vendor_key mentioned instead of just specifying the vendor_name.
Hence, the current PR restores the previous behavior wherein media settings can be retrieved with just vendor name.

#### How Has This Been Tested?

After xcvrd boot-up, it was ensured that the media_settings are updated in APPL_DB.
```
root@sonic:/var/log# cat swss/swss.rec | grep PORT_TABLE
2024-01-08.21:12:04.415172|PORT_TABLE:Ethernet188|SET|idriver:0x6,0x6,0x6,0x6
2024-01-08.21:12:15.887980|PORT_TABLE:Ethernet12|SET|idriver:0x6,0x6,0x6,0x6
```

Snippet of media_settings.json
```
root@str3-8111-03:/home/admin# cat /usr/share/sonic/device/x86_64-8111_32eh_o-r0/media_settings.json 
{
    "GLOBAL_MEDIA_SETTINGS": {
        "0-31": {
            "VENDORA           ": {
                "idriver": {
                    "lane0":"0x6",
                    "lane1":"0x6",
                    "lane2":"0x6",
                    "lane3":"0x6",
                    "lane4":"0x6",
                    "lane5":"0x6",
                    "lane6":"0x6",
                    "lane7":"0x6"
                }
            }
        }
    },
    "PORT_MEDIA_SETTINGS": {
    }
}
``
#### Additional Information (Optional)

### CODE REVIEW

**Comment by mihirpat1** [comment]
@tshalvi - It will be great if you can help in reviewing this.

File: sonic-xcvrd/xcvrd/xcvrd_utilities/media_settings_parser.py [hunk 1]
```diff
@@ -201,6 +201,14 @@ def get_media_settings_value(physical_port, key):
                         return {}
                 else:
                     return media_dict[key[VENDOR_KEY]]
+            elif key[VENDOR_KEY].split('-')[0] in media_dict:
```
Comment (tshalvi) [inline sonic-xcvrd/xcvrd/xcvrd_utilities/media_settings_parser.py:204]: The same code addition should be also added to JSONs with 'PORT_MEDIA_SETTINGS' section (right after line 245 in the updated code), and maybe a matching test case in test_xcvrd.py if you find it necessary.

Comment (mihirpat1) [reply sonic-xcvrd/xcvrd/xcvrd_utilities/media_settings_parser.py:204]: I have addressed this now and have added the relevant testcase as well.

**Review by prgeor — APPROVED** [review]

**Comment by prgeor** [comment]
@tshalvi can you review again?

**Review by tshalvi — APPROVED** [review]

**Pull request merged**